homeassistant/components/eafm/sensor.py

```diff
@@ -158,7 +170,33 @@ def unit_of_measurement(self):
     @property
     def device_state_attributes(self):
         """Return the sensor specific state attributes."""
-        return {ATTR_ATTRIBUTION: self.attribution}
+        return {
+            "typical_range_high": self.coordinator.data["stageScale"][0][
```

Comment:
`get_stage_scale()` can return `[]`, so `data["stageScale"]` can be `{}`, so this can fail with a `KeyError`. Please make sure the function doesn't explode if there is no stageScale.

Reply:
Also the wrapping here gets a bit ugly. Maybe have a temporary variable to get the stageScale out of the coordinator, and reuse that. E.g.

```
stage_scale = self.coordinator.data["stageScale"][0]

return {
...
highest_recent_date: stage_scale["highestRecent"]["dateTime"]
...
}
```